cppwinrt/code_writers.h

```diff
@@ -1129,17 +1129,27 @@ namespace cppwinrt
             if (is_remove_overload(method))
             {
                 // we intentionally ignore errors when unregistering event handlers to be consistent with event_revoker
+                //
+                // The `noexcept` versions will crash if check_cast_result throws but that is no different than previous
+                // behavior where it would not check the cast result and nullptr crash.  At least the exception will terminate
+                // immediately while preserving the error code and local variables.
                 format = R"(    template <typename D%> auto consume_%<D%>::%(%) const noexcept
     {%
-        WINRT_IMPL_SHIM(%)->%(%);%
+        auto const& castedResult = static_cast<% const&>(static_cast<D const&>(*this));
+        auto const abiType = *(abi_t<%>**)&castedResult;
+        check_cast_result(abiType);
```

Comment:
I would just point out that this check is done even when no QI has occurred. I think its reasonable as the check short-circuits quickly but its worth noting.

Reply:
That's fair.  In the common case where the cast succeeds (or there was no cast at all) it is a single nullptr check that gets inlined into the calling code.  This seems like a reasonable cost.  If evidence arises that this is too much overhead then we can always revisit (perhaps with an `if constexpr` to detect if the QI is going to be needed?).

Reply:
I would be surprised if a decent optimizer doesn't eliminate this.